### PR TITLE
feat: add mqtt v5 specific bindings to mqtt

### LINKED ISSUE #198: add message size/length to mqtt and mqtt5
#### Reason/Context
It would be valuable to add message size/length (in bytes) to the MQTT bindings to advertise the maximum message size a given MQTT binding accepts/emits.

#### Description
Given this is a feature/enhancement, it is not a breaking change.  Example definition:

MQTT5:
```yaml
    bindings:
      mqtt5:
        messageMaxSize: 4096
```

MQTT:
```yaml
    bindings:
      mqtt:
        messageMaxSize: 4096
```

Happy to put forth a PR if the above seems reasonable.

### LINKED ISSUE #199: MQTT5 CONNECT and PUBLISH Properties
#### Reason/Context
One of the defining features of MQTTv5 are _Properties_.  (otherwise, you'd probably just use MQTT v3).  Properties can be included in almost all MQTT packet types, and can form a significant part of an MQTTv5 API contract.  

In issue https://github.com/asyncapi/spec/issues/878 @fmvilas suggested that `bindings` are the appropriate construct for describing _properties_. 

For more information, this page lists properties by packet type: http://www.steves-internet-guide.com/mqttv5-properties/

#### Description
I suggest adding additional fields to the MQTT5 server binding for MQTT CONNECT packets:
 * Maximum Packet Size
 * User Property
 * Session Expiry Interval (already included in the MQTT5 0.2.0 binding)

I suggest adding additional fields to the MQTT5 message bindings for MQTT PUBLISH packets:
 * Paylod Format Indicator
 * Message Expiry Interval
 * Response Topic
 * Correlation Data
 * ~User Property~ (These are OK to be message headers per https://github.com/asyncapi/spec/issues/878#issuecomment-1328564400)
 * Content Type

I have a pull request that is about ready to go once this issue is approved (according to the contributing guidelines)

_Side note: I've used the word "packet" above when the correct MQTT terminology is "message" to disambiguate from AsyncAPI "message" objects._

### PR DESCRIPTION
**Pull request by pearmaster**
**Description**
According to https://github.com/asyncapi/bindings/issues/199#issuecomment-1567965696 , I'm only adding fields to the `mqtt` binding. This is because we don't want to create new sets of bindings for each version of each protocol.

Some general changes:
- Increased `mqtt` binding version to `0.2.0`
- Added a column to the tables to indicate which MQTT versions the binding applied to.
- Create a new **message** binding
- Added a deprecation warning to the `mqtt5` readme to indicate a consolidation of bindings under `mqtt` only.

 Here are specific fields being added:
- `sessionExpiryInterval` added to the **server** binding
- `maximumPacketSize` added to the  **server** binding
- `messageExpiryInterval` added to the **operation** binding
- `payloadFormatIndicator` added to the **message** binding
- `correlationData` added to the **message** binding
- `contentType` added to the **message** binding
- `responseTopic` added to the **message** binding.

**Related issue(s)**
Fixes #199 by adding new v5 properties to the mqtt binding
Fixes #198 by adding maximum packet size

### CODE REVIEW

**Review by fmvilas — APPROVED** [review]

**Comment by fmvilas** [comment]
@dalelane @derberg @smoya @char0n would love to know your opinion.

File: mqtt5/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 This document defines how to describe MQTT 5-specific information on AsyncAPI.
 
+**Deprecation Warning**: MQTT version 5 specific bindings are deprecated in favor of [MQTT bindings](../mqtt/README.md) that are not version specific.
```
Comment (derberg) [inline mqtt5/README.md:5]: The deprecation note is not very visible:

Screenshot 2023-06-26 at 19 40 38

```suggestion
# **Deprecation Warning**: MQTT version 5 specific bindings are deprecated in favor of [MQTT bindings](../mqtt/README.md) that are not version specific.
```

better make it `h1` and best probably move to the very top of the document

**Review by KhudaDad414 — APPROVED** [review]
Documentation seems to match the JSON schema specification. 🙇

File: mqtt5/README.md (hunk 1 above)
Comment (derberg) [reply mqtt5/README.md:5]: @pearmaster please have a look

**Review by derberg — APPROVED** [review]
lgtm

ping @fmvilas @smoya @dalelane @char0n

**Comment by smoya** [comment]
LGTM! 🚀🌔

Do we want to merge this into `master` and then adapt it to work with AsyncAPI v3? That already happened in https://github.com/asyncapi/bindings/pull/205, so we will need to update the binding in `next-major-spec` branch as well.
As an alternative, we could do the changes based on `next-major-spec` and will be released with v3 of the spec.

Opinions? cc @jonaslagoni @derberg

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

File: mqtt/README.md [hunk 2]
```diff
@@ -96,18 +136,48 @@ This object contains information about the message representation in MQTT.
...
 ##### Fixed Fields
 
-Field Name | Type | Description
----|:---:|---
-<a name="messageBindingObjectBindingVersion"></a>`bindingVersion` | string | The version of this binding. If omitted, "latest" MUST be assumed.
+Field Name | Type | MQTT Versions | Description
+---|:---:|:---:|---
+<a name="messageBindingObjectPayloadFormatIndicator"></a>`payloadFormatIndicator` | integer | 5 | Either: **0** (zero): Indicates that the payload is unspecified bytes, or **1**: Indicates that the payload is UTF-8 encoded character data. |
+<a name="messageBindingObjectCorrelationData"></a>`correlationData` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) | 5 | Correlation Data is used by the sender of the request message to identify which request the response message is for when it is received.
+<a name="messageBindingObjectContentType"></a>`contentType` | string | 5 | String describing the content type of the message payload. This should not conflict with the `contentType` field of the associated AsyncAPI Message object.
```
Comment (jonaslagoni) [inline mqtt/README.md:143]: How come the contentType of the AsyncAPI Message object is not enough 🤔

File: mqtt/README.md [hunk 3]
```diff
@@ -96,18 +136,48 @@ This object contains information about the message representation in MQTT.
 
 ##### Fixed Fields
 
-Field Name | Type | Description
----|:---:|---
-<a name="messageBindingObjectBindingVersion"></a>`bindingVersion` | string | The version of this binding. If omitted, "latest" MUST be assumed.
+Field Name | Type | MQTT Versions | Description
+---|:---:|:---:|---
+<a name="messageBindingObjectPayloadFormatIndicator"></a>`payloadFormatIndicator` | integer | 5 | Either: **0** (zero): Indicates that the payload is unspecified bytes, or **1**: Indicates that the payload is UTF-8 encoded character data. |
+<a name="messageBindingObjectCorrelationData"></a>`correlationData` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) | 5 | Correlation Data is used by the sender of the request message to identify which request the response message is for when it is received.
```
Comment (jonaslagoni) [inline mqtt/README.md:142]: How is this different from the correlation ID? 🤔

**Review by jonaslagoni — COMMENTED** [review]
Think there is reason to have this for v2 regardless of v3 is right around the corner.

Someone just needs to update these afterwards 😄

**Comment by derberg** [comment]
we should merge to master, so it is changed for v2 too

**Comment by jonaslagoni** [comment]
ping ping @pearmaster 🙂

File: mqtt/README.md (hunk 2 above)
Comment (jonaslagoni) [reply mqtt/README.md:143]: Got it 👍

File: mqtt/README.md (hunk 3 above)
Comment (jonaslagoni) [reply mqtt/README.md:142]: Alright 👍

**Review by char0n — APPROVED** [review]
LGTM

**Review by smoya — APPROVED** [review]
LGTM! 🚀🌔

**Comment by smoya** [comment]
> ping ping @pearmaster 🙂

I believe we can proceed with the merge of this PR. Nothing is expected from @pearmaster afaik. Anything else you expect @jonaslagoni ?

**Comment by jonaslagoni** [comment]
@smoya nothing but the JSON Schema files need to be moved to spec-json-schema repo 🙂

**Comment by GreenRover** [comment]
What is the status? 
I was about to do the same now, but have seen this pr. 
So i would be happy to see this merged.

**Pull request merged**